submissions/Oleksii_Drohachov/Memory_pair_game/script.js

```diff
@@ -0,0 +1,203 @@
+modalWrapper.append(modalBtn);
+modalWrapper.append(closeBtn);
+
+modal.append(modalWrapper);
+
+renderModal(
+  "You are welcome, my friend ! Wanna play some memory game to endure your mind and become more powerfull ? Try to find pairs of cards as fast as you can! Tap the button to begin... ",
+  "START THE GAME"
+);
+
+function renderModal(titleText, btnText) {
+  modalTitle.textContent = titleText;
+  modalBtn.textContent = btnText;
+  modal.classList.add("shown");
+}
+
+function hideModal() {
+  modal.classList.remove("shown");
+}
+
+function startGame() {
+  scoreCount.innerHTML = "";
+  gameboard.innerHTML = "";
+  hideModal();
+  prepareGame();
+}
+
+function prepareGame() {
+  const cardsArray = [
+    {
+      name: "beast",
+      path: "img/Beast.jpg",
+    },
+    {
+      name: "death",
+      path: "img/Death.jpg",
+    },
+    {
+      name: "fire",
+      path: "img/Fire.jpg",
+    },
+    {
+      name: "life",
+      path: "img/Life.jpg",
+    },
+    {
+      name: "light",
+      path: "img/Light.jpg",
+    },
+    {
+      name: "metal",
+      path: "img/Metal.jpg",
+    },
+  ];
+  const shuffled = [...cardsArray, ...cardsArray];
+  shuffled.sort(function () {
+    return 0.5 - Math.random();
+  });
+  let score = 0;
+  let guessedCardPairs = [];
+
+  renderScoreAndReset();
+  renderCards(shuffled);
+  cardsHandler();
+
+  const scoreSpan = document.querySelector(".score__span");
+
+  function renderScoreAndReset() {
+    const scoreSpan = document.createElement("span");
+    scoreSpan.classList.add("score__span");
+    scoreSpan.textContent = `Number of tries: ${score}`;
+    const resetBtn = document.createElement("button");
+    resetBtn.addEventListener("click", () =>
+      renderModal("Are you sure, you want to restart the game?", "RESET")
+    );
+    resetBtn.classList.add("reset__btn");
+    resetBtn.textContent = "RESTART";
+    scoreCount.append(scoreSpan);
+    scoreCount.append(resetBtn);
+  }
+
+  function updateScore() {
+    scoreSpan.textContent = `Number of tries: ${score}`;
+  }
+
+  function renderCards(arr) {
+    const cardWrapper = document.createElement("div");
+    cardWrapper.classList.add("card__wrapper");
+
+    arr.forEach((item) => {
+      const card = document.createElement("div");
+      card.classList.add("card");
+      card.setAttribute("data-name", item.name);
+      const frontPage = document.createElement("img");
+      frontPage.classList.add("front__page");
+      frontPage.setAttribute("src", item.path);
+      const backPage = document.createElement("img");
+      backPage.classList.add("back__page");
+      backPage.setAttribute("src", "img/backside.jpeg");
+
+      card.append(frontPage);
+      card.append(backPage);
+      cardWrapper.append(card);
+    });
+    gameboard.append(cardWrapper);
+  }
+
+  function cardsHandler() {
+    gameboard.addEventListener("click", ({ target }) => {
+      if (
+        target.parentElement.classList.contains("card") &&
```

Comment:
Don't use constructions like children[0], firstElementChild, parentNode, nextSibling, etc. In such way, you rely on the order of DOM elements. So in case when you will have changed design - your code will be broken. Which is bad, obviously. Use querySelector or closest, if in event, instead. [Read more](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest)